crates/wysiwyg/src/format_nodes.rs

```diff
@@ -0,0 +1,405 @@
+use crate::composer_model::{slice, slice_from, slice_to};
+use crate::dom::nodes::{ContainerNodeKind, DomNode, TextNode};
+use crate::dom::{
+    Dom, DomHandle, DomLocation, MultipleNodesRange, Range, SameNodeRange,
+};
+use crate::{ComposerModel, ComposerUpdate, InlineFormatType, UnicodeString};
+
+#[derive(Eq, PartialEq, Debug)]
+enum FormatSelectionType {
+    Extend,
+    Remove,
+}
+
+impl<S> ComposerModel<S>
```

Comment:
I'm a bit uncomfortable about providing more `impl` for `ComposerModel` in a different file, just because it might fool me later when I go looking for e.g. `format()` and don't find it in `composer_model.rs`. What do you think? I'd prefer free functions (maybe taking a `ComposerModel` as an argument, and annoyingly all having to have the generic arg specified), or a new struct like `NodeJoiner`, but am open to persuasion.

Reply:
Definitely wanted to talk about this in the sync. A struct with associated functions might be fine.

Otherwise what I would expect at least is having a composer_model dedicated crate/folder with just the files that are directly tied to it. Either the multiple `impl` or the structs with associated funcs. 
I'm not sure what would be the convention in Rust but if we go with multiple `impl` files I would expect us to change naming to be something like `composer_model+format.rs`, `composer_model+lists.rs`, etc then it would be easier to browse.

Reply:
I don't have a strong preference for one over the other, I just did it this way because it seemed simpler. I agree with @aringenbach though, if we do provide several impl we should probably group them in a crate so they're easy to discover.